src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -1013,4 +1013,21 @@ private boolean userHasPermissions(StudentParticipation participation) {
     public List<StudentParticipation> findWithSubmissionsWithResultByStudentIdAndExercise(Long studentId, Set<Exercise> exercises) {
         return studentParticipationRepository.findByStudentIdAndExerciseWithEagerSubmissionsResult(studentId, exercises);
     }
+
+    /**
+     * Get a mapping of participation ids to the number of submission for each participation.
+     *
+     * @param exerciseId the id of the exercise for which to consider participations
+     * @return the number of submissions per participation in the given exercise
+     */
+    public Map<Long, Integer> countSubmissionsPerParticipationByExerciseId(Long exerciseId) {
```

Comment:
Can you also change this `Long exerciseId` one to primitive type please?

Reply:
Done